### PR TITLE
Adds Community Browsing/Front Page View

### LINKED ISSUE #64: Community Browse Page
* We want a route to display the community browse page (`meme.place/c/<name>`)
* We want to allow the user to browse memes for that community with a tab at the top to change between top and new memes
* We want to display which community they are on in a bar at the top with a toolbar, with a button to see the details of the community in a dialog
* We want to be able to paginate memes within the browse

### PR DESCRIPTION
**Pull request by Step7750**
* Adds toolbar at the top to see community details, its name, favourite it, amount of favourites, and a button to show a dialog with the community's description and other text
* Restructures some services to aid in favouriting, UserService gets auto updated without having to make another request for user details.
* Shows badge next to name if NSFW
* Redirects to 404 if community is not found
* Adds paginator to paginate the shown memes
* Shows loading spinner while loading memes

Will test further when the server merges the PR for getting lists of memes

Closes #64

### CODE REVIEW

**Comment by Step7750** [comment]
@Dave0921, @bandgeekdante  I'd like your opinions before merging

File: src/app/community/browse/browse.component.ts [hunk 1]
```diff
@@ -1,16 +1,24 @@
...
 
-  constructor() { }
+  community: Community;
+  retrieved = false;
+
+  constructor(private route: ActivatedRoute,
+              private router: Router,
+              private communityService: CommunityService) { }
 
   user1 = {height: '300', memeId: 8};
```
Comment (Step7750) [inline src/app/community/browse/browse.component.ts]: These lines can be removed.

**Comment by Dave0921** [comment]
Okay yea I'm looking at the merge atm; there's a lot to go through haha.

**Comment by Dave0921** [comment]
Just wanted to confirm, it looks like we are sticking to using the star border (rather than solid star for favourited communities) for both favourited and unfavourited communities right? Other than that it is good.

**Comment by Step7750** [comment]
It uses a star border for unfavourited and star for favourited

`{{isFavourited ? 'star': 'star_border'}}`

**Comment by Dave0921** [comment]
Oh sorry missed that.

**Pull request merged**